tensor/stats/stats/funcs.go

```diff
@@ -4,62 +4,341 @@
+		if c == 0 {
+			continue
+		}
+		sum64.SetFloat1D(sum64.Float1D(i)/c, i)
+		out.SetFloat1D(sum64.Float1D(i), i)
+	}
+	return sum64, count64
+}
+
+// MeanFunc computes the mean of tensor values.
+// See [StatsFunc] for general information.
+func MeanFunc(in, out *tensor.Indexed) {
+	MeanFuncOut64(in, out)
+}
+
+// SumSqDevFuncOut64 computes the sum of squared mean deviates of tensor values,
+// and returns the Float64 output values for subsequent use.
+func SumSqDevFuncOut64(in, out *tensor.Indexed) (ssd64, mean64, count64 *tensor.Indexed) {
+	mean64, count64 = MeanFuncOut64(in, out)
+	ssd64 = VectorizeOut64(NFunc, func(idx int, tsr ...*tensor.Indexed) {
+		Vec2inFunc(idx, tsr[0], tsr[1], tsr[2], 0, func(val1, val2, agg float64) float64 {
+			dv := val1 - val2
+			return agg + dv*dv
+		})
+	}, in, mean64, out)
+	return
+}
+
+// VarFuncOut64 computes the sample variance of tensor values,
+// and returns the Float64 output values for subsequent use.
+func VarFuncOut64(in, out *tensor.Indexed) (var64, mean64, count64 *tensor.Indexed) {
+	var64, mean64, count64 = SumSqDevFuncOut64(in, out)
+	nsub := out.Tensor.Len()
+	for i := range nsub {
+		c := count64.Float1D(i)
+		if c < 2 {
+			continue
+		}
+		vr := var64.Float1D(i) / (c - 1)
+		var64.SetFloat1D(vr, i)
+		out.SetFloat1D(vr, i)
+	}
+	return
+}
+
+// VarFunc computes the sample variance of tensor values.
+// Squared deviations from mean, divided by n-1. See also [VarPopFunc].
+// See [StatsFunc] for general information.
+func VarFunc(in, out *tensor.Indexed) {
+	VarFuncOut64(in, out)
+}
+
+// StdFuncOut64 computes the sample standard deviation of tensor values.
+// and returns the Float64 output values for subsequent use.
+func StdFuncOut64(in, out *tensor.Indexed) (std64, mean64, count64 *tensor.Indexed) {
+	std64, mean64, count64 = VarFuncOut64(in, out)
+	nsub := out.Tensor.Len()
+	for i := range nsub {
+		std := math.Sqrt(std64.Float1D(i))
+		std64.SetFloat1D(std, i)
+		out.SetFloat1D(std, i)
+	}
+	return
+}
+
+// StdFunc computes the sample standard deviation of tensor values.
+// Sqrt of variance from [VarFunc]. See also [StdPopFunc].
+// See [StatsFunc] for general information.
+func StdFunc(in, out *tensor.Indexed) {
+	StdFuncOut64(in, out)
+}
+
+// SemFunc computes the sample standard error of the mean of tensor values.
+// Standard deviation [StdFunc] / sqrt(n). See also [SemPopFunc].
+// See [StatsFunc] for general information.
+func SemFunc(in, out *tensor.Indexed) {
+	var64, _, count64 := VarFuncOut64(in, out)
+	nsub := out.Tensor.Len()
+	for i := range nsub {
+		c := count64.Float1D(i)
+		if c < 2 {
+			out.SetFloat1D(math.Sqrt(var64.Float1D(i)), i)
+		} else {
+			out.SetFloat1D(math.Sqrt(var64.Float1D(i))/math.Sqrt(c), i)
+		}
+	}
+}
+
+// VarPopFuncOut64 computes the population variance of tensor values.
+// and returns the Float64 output values for subsequent use.
+func VarPopFuncOut64(in, out *tensor.Indexed) (var64, mean64, count64 *tensor.Indexed) {
```

Comment:
There is a question as to whether you would be better off spelling out these names some more...